account_invoice_tax_note/models/account_invoice.py

```diff
@@ -0,0 +1,26 @@
+# Copyright 2018 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+from odoo import api, exceptions, models, _
+
+
+class AccountInvoice(models.Model):
+    _inherit = 'account.invoice'
+
+    @api.multi
+    def _get_account_tax_groups(self):
+        """
+        Get every account.tax.group used on current recordset where
+        report_note is filled.
```

Comment:
I suggest renaming `_get_account_tax_groups_with_notes()` (and removing the docstring comment which become redundant with the method name).

Because in the template you still read a call to _get_account_tax_groups and you don't immediately understand there is a filtering behind.